### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by VeraLapidus**

### CODE REVIEW

File: auto_show/tests.py [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+    return AutoShow.objects.create(
+        name="Ford",
+        country="DE",
+        balance=200.00,
+        owner=User.objects.get(username="Ford Minsk"))
+
+
+@pytest.mark.django_db
+def test_user_create(create_user):
+    assert User.objects.filter(username="Ford Minsk").exists()
```
Comment (shprexen) [inline auto_show/tests.py]: We don't test Django functions
We should focus on implemented business logic

**Review by shprexen — CHANGES_REQUESTED** [review]
In general - you've done it pretty good
Add tests for celery tasks logic, as well for stats endpoints
I recommend you to use `coverage` tool, to see how many rows of code have been covered by tests. We should aim for 90 percent coverage

File: auto_show/tests.py [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+@pytest.mark.django_db
+def test_auto_show_delete(create_auto_show):
+    AutoShow.objects.filter(name="Ford").delete()
+    assert AutoShow.objects.count() == 0
+
+
+@pytest.mark.django_db
+def test_auto_show_statistic(create_auto_show, api_client):
+    url = reverse('autoshow-stat', kwargs={'pk': create_auto_show.id})
+    response = api_client.get(url, pk=create_auto_show.id)
```
Comment (shprexen) [inline auto_show/tests.py:64]: test, if response body is valid

**Review by shprexen — CHANGES_REQUESTED** [review]

File: auto_show/tests.py (hunk 2 above)
Comment (VeraLapidus) [reply auto_show/tests.py:64]: done

**Review by shprexen — CHANGES_REQUESTED** [review]
check authentication of endpoints, add requests for unauthorized actions to check 403 status code

File: deals/tests.py [hunk 3]
```diff
@@ -101,3 +108,12 @@ def test_celery_find_best_car_for_sale(create_auto_show_for_celery, create_car_i
...
     assert find_best_car_for_sale.delay().get(timeout=10) == '15.00'
+
+
+@pytest.mark.django_db
+def test_authenticate_but_forbidden_request(api_client, create_user_auto_show):
+    user = User.objects.get(username="Audi AutoShow")
+    api_client.force_authenticate(user=user)
+    url = reverse('deal-list')
+    response = api_client.get(url)
+    assert response.status_code == 403
```
Comment (shprexen) [inline deals/tests.py]: use builtin status codes `from rest_framework import status`, for example: `status.HTTP_404_NOT_FOUND`
According to DRF docs:
"Using bare status codes in your responses isn't recommended. REST framework includes a set of named constants that you can use to make your code more obvious and readable."

**Review by shprexen — CHANGES_REQUESTED** [review]

**Review by shprexen — APPROVED** [review]

**Pull request merged**